qp_target_gene/pick_otus.py

```diff
@@ -67,10 +67,16 @@ def generate_pick_closed_reference_otus_cmd(filepaths, out_dir, parameters,
     write_parameters_file(param_fp, parameters)
 
     cmd_ungz = ''
-    if not test and is_zipfile(seqs_fp):
-        seqs_fp_fna = join(out_dir, 'seqs.fna')
-        cmd_ungz = 'gunzip -c %s > %s && ' % (seqs_fp, seqs_fp_fna)
-        seqs_fp = seqs_fp_fna
+    if not test:
+        try:
+            with gopen(seqs_fp, 'rb') as f:
+                f.read(1)
+            seqs_fp_fna = join(out_dir, 'seqs.fna')
+            cmd_ungz = 'pigz -c -d -p%s %s > %s && ' % (
+                str(parameters['threads']), seqs_fp, seqs_fp_fna)
+            seqs_fp = seqs_fp_fna
+        except (OSError, IOError):
+            pass
```

Comment:
Not sure I follow what's going on with this try/except block. What do you think about checking for the magic number like it was [done in QIIME1](https://github.com/biocore/qiime/blob/master/qiime/util.py#L1421)? Seems too me like it isn't a good idea to have a multi-line try/except block.

Aside from that, do we need `-d`? I was just reading the man page and the flag's help says "decompress the compressed input":
https://zlib.net/pigz/pigz.pdf

```suggestion
    if not test:
        with read(seqs_fp, 'rb') as f:
            is_gzipped = f.read(2) == b'\x1f\x8b'
        if is_gzipped:
            seqs_fp_fna = join(out_dir, 'seqs.fna')
            cmd_ungz = 'pigz -c -d -p %s %s > %s && ' % (
                str(parameters['threads']), seqs_fp, seqs_fp_fna)
            seqs_fp = seqs_fp_fna
```

Reply:
- check for gz: Checking for the magic number is an option and I'm OK to change it if you consider it better than the try/except suggestion. I thought try/except was better based on this comment: https://stackoverflow.com/a/3703300 - aka not 100% reliable
- -d: it's to decompress; I can change to unpigz if you think is bette

Reply:
- I see, in that case if we can remove the multi-line try/except that would probably be the only thing to do.
- Gotcha, leaving I got confused by the command name -d is fine or using unpigz. Whichever way you prefer.